functions/pupil-status/v1.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+const pupilStatusService = require('./pupil-status.service')
+
+async function process (context, pupilStatusMessage) {
+  if (!pupilStatusMessage.hasOwnProperty('pupilId')) {
```

Comment:
@jon-shipley isn't that essentially a validator?
I did extract validation separately into a file on some other functions.
Perhaps the same approach should be followed for consistency?

Reply:
@GeorgeChatzigiannis This is a lift & shift of existing code to prove a performance driven change.  I would rather not change too much in this PR

Reply:
@GuyHarwood it would make the validation isolated and easily testable which is the main reason why I did it on other functions.
It can certainly happen after this PR is merged.